2996_reflection/reflection.md

```diff
@@ -4972,6 +5171,22 @@ Defines `class_type` with properties as follows:
 :::
 :::
 
+### [meta.reflection.static_string] Static string generation  {-}
+
+::: std
+::: addu
+```cpp
+consteval const char *define_static_string(std::string_view str);
+consteval const char8_t *u8define_static_string(std::u8string_view str);
+```
+
+[#]{.pnum} *Returns*: A pointer to the first element of an array of `char` or `char8_t` objects, respectively, whose length is `str.size() + 1`. The values of the first `str.size()` array subobjects are `str[k]`, and the value of the last array subobject is `0`. The array has static storage duration, and is usable in constant expressions.
+
+[#]{.pnum} [Implementations are encouraged to return the same pointer whenever the same variant of these functions is called with the same argument.]{.note}
```

Comment:
```cpp
constexpr char const* a = define_static_string("ghost");
constexpr char const* b = define_static_string("host");
```
Is it allowed for `a + 1 == b`?

Reply:
No, it isn't - But I think that this falls out from the requirement that the returned string is null-terminated.

Reply:
`"host"` is a suffix of `"ghost"`, with the null terminator. 

The way it's worded right now the answer is no just because we say it's the pointer to the first element of an array. Wonder if that's worth relaxing. Probably not?

Reply:
Oooo great point. Yeah, I guess it can probably always be relaxed later. But here's one way we might say it instead, let me know if you'd like to pursue:

> Let `S` be a constexpr variable of array type with static storage duration, whose elements are of type `const char` or `const char8_t` respectively, for which there exists some `k >= 0` such that:
> - `S[k+i] == str[i]` for all 0 <= `i` < `str.size()`, and
> - `S[k + str.size()] == '\0'`.
>
> _Returns_: `&S[k]`

@brevzin What do you think? Go for it, or leave as is?

Reply:
I like it!

Reply:
Done!